lib/active_merchant/billing/gateways/beanstream.rb

```diff
@@ -64,149 +65,29 @@ module Billing #:nodoc:
-        commit(post)
-      end
-
-      def update_recurring(amount, source, options = {})
-        ActiveMerchant.deprecated RECURRING_DEPRECATION_MESSAGE
-
-        post = {}
-        add_recurring_amount(post, amount)
-        add_recurring_invoice(post, options)
-        add_credit_card(post, source)
-        add_address(post, options)
-        add_recurring_operation_type(post, :update)
-        add_recurring_service(post, options)
-        recurring_commit(post)
-      end
-
-      def cancel_recurring(options = {})
-        ActiveMerchant.deprecated RECURRING_DEPRECATION_MESSAGE
-
-        post = {}
-        add_recurring_operation_type(post, :cancel)
-        add_recurring_service(post, options)
-        recurring_commit(post)
-      end
-
-      def interac
-        @interac ||= BeanstreamInteracGateway.new(@options)
-      end
-
-      # To match the other stored-value gateways, like TrustCommerce,
-      # store and unstore need to be defined
-      def store(payment_method, options = {})
-        post = {}
-        add_address(post, options)
-
-        if payment_method.respond_to?(:number)
-          add_credit_card(post, payment_method)
-        else
-          post[:singleUseToken] = payment_method
-        end
-        add_secure_profile_variables(post, options)
-
-        commit(post, true)
-      end
-
-      #can't actually delete a secure profile with the supplicated API. This function sets the status of the profile to closed (C).
-      #Closed profiles will have to removed manually.
-      def delete(vault_id)
-        update(vault_id, false, {:status => 'C'})
-      end
-
-      alias_method :unstore, :delete
-
-      # Update the values (such as CC expiration) stored at
-      # the gateway.  The CC number must be supplied in the
-      # CreditCard object.
-      def update(vault_id, payment_method, options = {})
-        post = {}
-        add_address(post, options)
-        if payment_method.respond_to?(:number)
-          add_credit_card(post, payment_method)
-        else
-          post[:singleUseToken] = payment_method
-        end
-        options.merge!({:vault_id => vault_id, :operation => secure_profile_action(:modify)})
-        add_secure_profile_variables(post,options)
-        commit(post, true)
-      end
-
-      def supports_scrubbing?
-        true
-      end
-
-      def scrub(transcript)
-        transcript.
-          gsub(%r((Authorization: Basic )\w+), '\1[FILTERED]').
-          gsub(/(&?password=)[^&\s]*(&?)/, '\1[FILTERED]\2').
-          gsub(/(&?passcode=)[^&\s]*(&?)/, '\1[FILTERED]\2').
-          gsub(/(&?trnCardCvd=)\d*(&?)/, '\1[FILTERED]\2').
-          gsub(/(&?trnCardNumber=)\d*(&?)/, '\1[FILTERED]\2')
-      end
+    #
+    # Modification on 2018-03-12:
+    # New feature: integrated the ipp gateway into beanstream gateway, use this class as an unified entrance for both the gateways
+    # Get the :region field from options parameter, :region: 0 - US & CA, 1 - AU
```

Comment:
comment no longer reflects the code, symbol should be passed instead of integers

Reply:
Fixed.